crates/egui/src/containers/window.rs

```diff
@@ -429,6 +429,12 @@ impl<'open> Window<'open> {
             (0.0, 0.0)
         };
 
+        let viewport_rect = ctx.input(|i| i.viewport().inner_rect.unwrap_or(Rect::EVERYTHING));
+        let max_width = viewport_rect.width();
+        let max_height = viewport_rect.height() - title_bar_height;
+        resize.max_size.x = resize.max_size.x.min(max_width);
+        resize.max_size.y = resize.max_size.y.min(max_height);
```

Comment:
what I meant was this:

```suggestion
        let max_rect = area.constrain_rect.unwrap_or_else(|| ui.ctx().screen_rect);
        let max_width = max_rect.width();
        let max_height = max_rect.height() - title_bar_height;
        resize.max_size.x = resize.max_size.x.min(max_width);
        resize.max_size.y = resize.max_size.y.min(max_height);
```